lib/api/unittest/CDataFrameAnalyzerTrainingTest.cc

```diff
@@ -2217,4 +2218,83 @@ BOOST_AUTO_TEST_CASE(testProgressMonitoringFromRestart) {
+        test::CDataFrameAnalysisSpecificationFactory specFactory;
+        return specFactory.rows(rows)
+            .columns(6)
+            .memoryLimit(15000000)
+            .predictionMaximumNumberTrees(3)
+            .predictionPersisterSupplier(persisterSupplier)
+            .predictionRestoreSearcherSupplier(restorerSupplier)
+            .regressionLossFunction(TLossFunctionType::E_MseRegression)
+            .task(task)
+            .predictionSpec(test::CDataFrameAnalysisSpecificationFactory::regression(),
+                            "target", &frameAndDirectory);
+    };
+
+    TStrVec fieldNames{"c1", "c2", "c3", "c4", "c5", "target", ".", "."};
+    TStrVec fieldValues{"", "", "", "", "", "", "0", ""};
+    TDataFrameUPtrTemporaryDirectoryPtrPr frameAndDirectory;
+
+    std::stringstream output;
+    auto outputWriterFactory = [&output]() {
+        return std::make_unique<core::CJsonOutputStreamWrapper>(output);
+    };
+
+    auto persistenceStream = std::make_shared<std::ostringstream>();
+    TPersisterSupplier persisterSupplier{[&persistenceStream]() {
+        return std::make_unique<api::CSingleStreamDataAdder>(persistenceStream);
+    }};
+    auto spec = makeSpec(rowsEncode, frameAndDirectory, &persisterSupplier, nullptr,
+                         test::CDataFrameAnalysisSpecificationFactory::TTask::E_Encode);
+    api::CDataFrameAnalyzer analyzer{std::move(spec), std::move(frameAndDirectory),
+                                     outputWriterFactory};
+    test::CDataFrameAnalyzerTrainingFactory::addPredictionTestData(
+        TLossFunctionType::E_MseRegression, fieldNames, fieldValues, analyzer, rowsEncode);
+    analyzer.handleRecord(fieldNames, {"", "", "", "", "", "", "", "$"});
+
+    TStrVec persistedStates{
+        splitOnNull(std::stringstream{std::move(persistenceStream->str())})};
+
+    BOOST_REQUIRE(persistedStates.size() > 0);
+
+    // pass persisted state as a state to restore from in the new analyzer
+    std::istringstream lastStateStream{persistedStates[0]};
+    TRestoreSearcherSupplier restoreSearcherSupplier{[&lastStateStream]() {
+        return std::make_unique<CTestDataSearcher>(lastStateStream.str());
+    }};
+    spec = makeSpec(rowsTrain, frameAndDirectory, nullptr, &restoreSearcherSupplier,
+                    test::CDataFrameAnalysisSpecificationFactory::TTask::E_Train);
+    api::CDataFrameAnalyzer restoredAnalyzer{
+        std::move(spec), std::move(frameAndDirectory), outputWriterFactory};
+
+    test::CDataFrameAnalyzerTrainingFactory::addPredictionTestData(
+        TLossFunctionType::E_MseRegression, fieldNames, fieldValues,
+        restoredAnalyzer, rowsTrain);
+    restoredAnalyzer.handleRecord(fieldNames, {"", "", "", "", "", "", "", "$"});
+
+    // reset logger for next tests
+    logger.reset();
+
+    // Check that no error messages were logged
+    BOOST_TEST_REQUIRE(logDataSPtr->rdbuf()->in_avail() == 0);
```

Comment:
Should we perhaps assert something about the predictions, i.e. at least that we have reasonable predictions written?

Reply:
I test the predictions in `CBoostedTreeTest`. At the api level, I simply check if the process works successfully when triggered from the outside.

Reply:
Ok, that seems reasonable.